### PR TITLE
feat: adds type gen for graphql and departure example

### PR DESCRIPTION
**Pull request by mikaelbr**
Adds integration to JourneyPlanner for getting stop places, nearest, and initial departures.

## Todo

- [x] Map types properly for external use

Related to https://github.com/AtB-AS/kundevendt/issues/9148

---

Showing how to fetch data from GraphQL with code generation

```ts
export function createJourneyApi(
  client: GraphQlRequester<'graphql-journeyPlanner3'>,
): JourneyPlannerApi {
  const api: JourneyPlannerApi = {
    async departures(input) {
      const result = await client.query<
        StopPlaceQuayDeparturesQuery,
        StopPlaceQuayDeparturesQueryVariables
      >({
        query: StopPlaceQuayDeparturesDocument,
        variables: {
          id: input.id,
          startTime: new Date(),
          numberOfDepartures: 10,
        },
      });

      if (result.error) {
        throw result.error;
      }

      return result.data;
    },
  };

  return api;
}
```

### CODE REVIEW

File: graphql.config.ts
Comment (adriansberg) [inline graphql.config.ts:1]: Could you convert this to yaml/.graphqlrc/json? My IDE didn't like TypeScript.

Comment (mikaelbr) [reply graphql.config.ts:1]: Yep! Forgot about that

File: package.json [hunk 1]
```diff
@@ -50,9 +59,11 @@
     "@vitejs/plugin-react": "^4.0.4",
     "eslint": "8.49.0",
     "eslint-config-next": "13.5.1",
+    "graphql-tag": "^2.12.6",
     "happy-dom": "^12.1.5",
     "prettier": "^3.0.3",
     "rimraf": "^5.0.1",
+    "vite": "^4.4.9",
```
Comment (adriansberg) [inline package.json:66]: What's `vite` used for?

File: src/modules/api-server/index.ts [hunk 2]
```diff
@@ -13,12 +13,11 @@ import { ExternalClient, ExternalClientFactory } from './external-client';
 export type {
   AllEndpoints,
   HttpRequester,
-  GraphQlRequester,
   ExternalClient,
   ExternalClientFactory,
 };
 
-export { ApplicationError };
+export { GraphQlRequester, ApplicationError };
```
Comment (adriansberg) [inline src/modules/api-server/index.ts:20]: Why was this moved?

File: package.json (hunk 1 above)
Comment (mikaelbr) [reply package.json:66]: It was a warning for a peer dependency of something I can't remember right now.

File: src/modules/api-server/index.ts (hunk 2 above)
Comment (mikaelbr) [reply src/modules/api-server/index.ts:20]: It's changed from not only being a type that is exported but also the class / constructor

**Review by adriansberg — APPROVED** [review]
Looking good 👍
